source/contribute/maintainer.rst

```diff
@@ -57,3 +57,29 @@ the ``use-next-minor-as-default-branch`` switch for the
     $ git switch --create 2.0.x 2.0.0
     Switched to a new branch "2.0.x"
     $ git push doctrine 2.0.x
+
+Tagging a release
+-----------------
+
+We use ``laminas/automatic-releases`` to make releasing as easy as
+closing a milestone. Before closing a milestone, one should check that
+it is complete. If not, they should unassign issues and pull requests,
+or close them / merge them as appropriate. It is also a good idea to
+check which pull requests were merged since the last release on the
+relevant branch, and to make sure each one is properly labelled and
+assigned to the milestone.
+
+Labels are used to generate the release notes, and we should strive to
+provide a consistent experience accross repositories. Here are labels
+that we recommend adding on every repository:
+
+Documentation
+  Includes changes to rst files, markdown files as well as code comments
+  that are not phpdoc comments.
```

Comment:
Not sure how to label a PR that changes phpdoc comments… "Static Analysis"?

Reply:
phpdoc comments become more and more obsolete and are replaced with type casts when it comes to types. But the descriptive content would still be a documentation. I don't know if we should focus more on phpstan/psalm annotations when it comes to types (and therefore "Static Analysis") and regular phpdocs when it's anout documentation. I don't know about possible grey lines though.

Reply:
Agree, the description part would still be documentation. Maybe the distinction will be too hard to make.

Reply:
IMO, "Static Analysis" has been used for the issues on detecting the issues in the API (e.g. by upgrading a dependency) and fixing the API contracts. A change in the API (native parameter/return type) or in the API _documentation_ (phpDoc) that isn't needed to address a static analysis build failure doesn't fall under the "static analysis" category IMO.

I don't know if it is important to distinguish the API and API documentation categories. On the one end, they both are about the API but on the other, a change in the API may be breaking while a change in the documentation cannot.

Reply:
A change in (API) documentation can be breaking for our users' CI, can't it? IMO, in that case, there is an impact, and since we are focusing on impact with this label, maybe we should prefer "Bugfix" / "Improvement" for those?

Reply:
No, a change in documentation cannot be breaking.

Reply:
Can you please elaborate? If we can make the CI of our user turn red, wouldn't you call that breaking?

Reply:
Let's assume the following library code:
```php
/**
 * @param string $string
 */
private function acceptsString($string);

/**
 * @param string|null $value
 */
public function publicFunction($value) {
    acceptsString($value);
}
```
And the client code:
```php
/**
 * @param string|null $value
 */
function doStuff($value) {
    publicFunction($value);
}
```
There is a bug in the `publicFunction()` documentation that it accepts `null` while it doesn't. The `null` case isn't covered by a test, and the CI on the library side isn't strict enough to identify the issue. At the same time. the CI on the client side is strict enough but there's no issue so far.

Now, the library maintainers decide to fix the documentation and change the API documentation to:
```php
/**
 * @param string $value
 */
public function publicFunction($value)
```
By doing that, they are not changing the API in any way that would break an existing production scenario. A production scenario that might rely on `publicFunction()` accepting `null` couldn't exist. But the client's CI will fail and point to a bug in their API documentation. Nothing more.

The impact of breaking CI by changing API documentation is the same as breaking CI that doesn't allow deprecations by introducing a deprecation.

Reply:
> The impact of breaking CI by changing API documentation is the same as breaking CI that doesn't allow deprecations by introducing a deprecation (which is also part of API documentation).

It sure is smaller, but as you wrote yourself, there is an impact. If the issue is accountability, we could still acknowledge that impact without making it part of our backwards compatibility promise. Then I guess it would not be "breaking"? To me, changes in phpdoc fall into 2 categories:
- more precise: "Improvement" (for static analysis)
- more accurate: "Bugfix" (for static analysis)

> IMO, "Static Analysis" has been used for the issues on detecting the issues in the API (e.g. by upgrading a dependency) and fixing the API contracts as a result of static analysis. In my understanding, this label was originally meant to justify the extra burden of running static analysis tools by exposing the issues it allowed to discover.

Right now we have 2 Static Analysis tools, so although we primarily act on our phpdoc "as a result of static analysis", doing so for one tool improves the experience when using the other, because it becomes able to detect new issues when we are more precise. To me, it might be worth reusing that "Static Analysis" label so that we make another section, which would sit between "Internal" and the rest. Would that make sense?

If we go that way, we should probably not label such PRs with "Improvement" or "Bugfix" so as to avoid having subsections since we cannot generate those yet.